11-walkthrough-7.Rmd

```diff
@@ -1,209 +1,311 @@
+
+First, let's make a dataset of tweets that had positive words from the NRC dataset. Remember earlier when we made a dataset of tweets that had "dataviz" and a column that had a value of 1 for containing positive words and 0 for not containing positive words? Let's resuse that technique, but instead of applying to a dataset of tweets containing "dataviz", let's use it on our dataset of all tweets. 
+
 ```{r positive tweets}
-tweets_pos <- tweets %>% 
-  mutate(pos_words = text %>% map_lgl(detect_pos)) 
+pos_tweets <- tweets %>% 
+  mutate(positive = if_else(status_id %in% unique(pos_tokens$status_id), 1, 0)) %>% 
+  filter(positive == 1)
 ```
 
-We'll use `slice`.  
+Again, we're using `if_else` to make a new column called `positive` that takes its value based on whether `status_id %in% unique(pos_tokens$status_id` is true or not. 
+
+We can use `slice` to help us pick the rows. When we pass `slice` a row number, it returns that row from the dataset. For example, we can select the 1st and 3rd row of our tweets datset this way: 
+
+```{r first and third tweet}
+tweets %>% 
+  slice(1, 3)
+```
+
+Randomly selecting rows from a dataset is great technique to have in your toolkit. Random selection helps us avoid some of the biases we all have when we pick rows to review ourselves. If we want to specifiy random row numbers for slice to pick for us, we can use `sample(1:nrow(.), 10)` like this: 
 
 ```{r randomly review}
 set.seed(369) 
 
-tweets_pos %>% 
-  filter(pos_words == TRUE) %>% 
-  slice(., sample(1:nrow(.), 10))
+pos_tweets %>% 
+  slice(., sample(x = 1:nrow(.), size = 10))
 ```
 
-# Next steps 
+That returned ten randomly selected tweets that we can now read through and discuss. Let's look a little closer at how we did that. We used a function called `sample` that returns randomly selected numbers from a vector. We passed the argument `1:nrow(.)` to `sample`, which evaluates as any number between 1 and the number of rows in `pos_tweets`. We also specified that `size = 10`, which means we want sample to give us 10 randomly selected numbers from `1:nrow(.)`. A few lines before that, we used `set.seed(369)`. This helps us ensure that, while `sample` theoretically plucks 10 random numbers, we want our readers to run this code and get the same result we did. Using `set.seed(369)` at the top of your code makes `sample` pick the same ten rows every time. Trying changing `369` to another number and notice how `sample` picks a different set of ten numbers, but repeatedly picks those numbers until you change the argument in `set.seed()`. 
+
+## Next steps 
```

Comment:
Perhaps for further reading - I think this will be open access (it's not published yet, but should be by the time we go to press with the boom):

Greenhalgh, S. P., Staudt Willet, B., Rosenberg, J. M., & Koehler, M. J. (accepted pending minor revisions). In E. Romero-Hall (Ed.), Research Methods in Learning Design & Technology., Lessons learned from applying Twitter research methods to educational technology phenomena. Routledge: New York, NY.